internal/http/client.go

```diff
@@ -47,66 +53,90 @@ func (c Client) AppendDefaultHeaders(headers RequestHeaders) RequestHeaders {
 	return headers
 }
 
+func (c Client) retryDuration(retry int) time.Duration {
+	duration := c.retryBaseDuration * (1 << retry)
+	if duration > c.retryMaxDuration {
+		return c.retryMaxDuration
+	}
+
+	duration += time.Duration(rand.Float64() * float64(time.Second))
```

Comment:
Should probably still add the jitter regardless (even if we're over max duration), it's max a second over anyway